### PR TITLE
Eventhubs ARM template for live test

### PR DESCRIPTION
**Pull request by hemanttanwar**
Our goal is to create common Azure resources using ARM template for each module. And pipeline to manage common resources and drop them in end. This will save cost and remove unwanted resources after test is complete or stopped in between
The pipeline will run ARM template and create common resources for example resource group, Accounts etc.
The pipeline will pass on names of these resources as ENV variable into test. For example AZURE_EVENTHUBS_CONNECTION_STRING
In the end pipeline will also delete these common resources which it created earlier and that will cause all the resources created to be removed.

### CODE REVIEW

File: sdk/eventhubs/test-resources.json [hunk 1]
```diff
@@ -0,0 +1,351 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
```
Comment (srnagar) [inline sdk/eventhubs/test-resources.json]: Is this expected to be under `sdk/eventhubs`? Should this be under a specific module? For e.g. `sdk/eventhubs/azure-messaging-eventhubs`?

Comment (hemanttanwar) [reply sdk/eventhubs/test-resources.json]: This is right place based on current strategy as we have already done in storage also.

**Comment by hemanttanwar** [comment]
@conniey    please look at this  pipeline run , I think it is now in a place to merge to master.

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=234963&view=logs&j=27eb5d14-7214-5d9c-6bac-653552bd18b3&t=8c9b85f5-3240-5af8-b929-e53e815dfe57

File: sdk/eventhubs/test-resources.json [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        "STORAGE_DATA_LAKE_ACCOUNT_NAME": {
+            "type": "string",
+            "value": "[variables('dataLakeAccountName')]"
+        },
+        "STORAGE_DATA_LAKE_ACCOUNT_KEY": {
+            "type": "string",
+            "value": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('dataLakeAccountName')), variables('storageApiVersion')).keys[0].value]"
+        },
+
+        "AZURE_STORAGE_BLOB_CONNECTION_STRING": {
```
Comment (conniey) [inline sdk/eventhubs/test-resources.json]: nit: we should try to keep these in alphabetical order.

File: sdk/eventhubs/test-resources.json [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+                        }
+                    },
+                    "keySource": "Microsoft.Storage"
+                },
+                "accessTier": "Hot"
+            }
+        }
+    ],
+    "outputs": {
+        "AZURE_EVENTHUBS_CONNECTION_STRING": {
```
Comment (conniey) [inline sdk/eventhubs/test-resources.json]: We have a test that tries to authenticate using `azure-identity` . Does this also output the AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, and AZURE_TENANT_ID?

Comment (hemanttanwar) [reply sdk/eventhubs/test-resources.json]: Yes we are getting these env properties .
https://github.com/Azure/azure-sdk-for-java/pull/7344/files#diff-8207928b799f9b8433d6601cec51886fR14

File: sdk/eventhubs/tests.yml [hunk 4]
```diff
@@ -4,21 +4,22 @@ jobs:
...
-        PREMIUM_STORAGE_ACCOUNT_KEY: $(java-storage-test-premium-account-key)
+        PRIMARY_STORAGE_ACCOUNT_NAME: $(PRIMARY_STORAGE_ACCOUNT_NAME)
+        PRIMARY_STORAGE_ACCOUNT_KEY: $(PRIMARY_STORAGE_ACCOUNT_KEY)
+        SECONDARY_STORAGE_ACCOUNT_NAME: $(SECONDARY_STORAGE_ACCOUNT_NAME)
+        SECONDARY_STORAGE_ACCOUNT_KEY: $(SECONDARY_STORAGE_ACCOUNT_KEY)
+        BLOB_STORAGE_ACCOUNT_NAME: $(BLOB_STORAGE_ACCOUNT_NAME)
+        BLOB_STORAGE_ACCOUNT_KEY: $(BLOB_STORAGE_ACCOUNT_KEY)
+        PREMIUM_STORAGE_ACCOUNT_NAME: $(PREMIUM_STORAGE_ACCOUNT_NAME)
+        PREMIUM_STORAGE_ACCOUNT_KEY: $(PREMIUM_STORAGE_ACCOUNT_KEY)
         AZURE_STORAGE_ACCESS_KEY: $(java-storage-test-access-key)
```
Comment (danieljurek) [inline sdk/eventhubs/tests.yml]: Is AZURE_STORAGE_ACCESS_KEY still needed?

File: sdk/eventhubs/tests.yml [hunk 5]
```diff
@@ -4,21 +4,22 @@ jobs:
...
     parameters:
       ServiceDirectory: eventhubs
+      TimeoutInMinutes: 120
       EnvVars:
-        AZURE_EVENTHUBS_CONNECTION_STRING: $(java-event-hubs-test-connection-string)
-        AZURE_EVENTHUBS_FULLY_QUALIFIED_DOMAIN_NAME: $(java-event-hubs-fully-qualified-domain-name)
-        AZURE_EVENTHUBS_EVENT_HUB_NAME: $(java-event-hubs-event-hub-name)
+        AZURE_EVENTHUBS_CONNECTION_STRING: $(AZURE_EVENTHUBS_CONNECTION_STRING)
+        AZURE_EVENTHUBS_FULLY_QUALIFIED_DOMAIN_NAME: $(AZURE_EVENTHUBS_FULLY_QUALIFIED_DOMAIN_NAME)
+        AZURE_EVENTHUBS_EVENT_HUB_NAME: $(AZURE_EVENTHUBS_EVENT_HUB_NAME)
```
Comment (danieljurek) [inline sdk/eventhubs/tests.yml]: Using the current version of New-TestResources.ps1 (https://github.com/Azure/azure-sdk-for-java/blob/master/eng/New-TestResources.ps1#L242) the variable mapping is no longer required for output variables. If you plumb all of the secrets through the ARM template you can reduce the `EnvVars` section down to: 

```
EnvVars:
    AZURE_TEST_MODE: RECORD
    AZURE_LOG_LEVEL: 4
```

You'll need to plumb the `AZURE_CLIENT_ID` and friends through the ARM template. 

```
...
"parameters": { 
        "tenantId": {
            "type": "string",
            "defaultValue": "72f988bf-86f1-41af-91ab-2d7cd011db47",
            "metadata": {
                "description": "The tenant ID to which the application and resources belong."
            }
        },
        "testApplicationId": {
            "type": "string",
            "metadata": {
                "description": "The application client ID used to run tests."
            }
        },
        "testApplicationSecret": {
            "type": "string",
            "metadata": {
                "description": "The application client secret used to run tests."
            }
        }, 
...

"outputs": { 
        "AZURE_TENANT_ID": {
            "type": "string",
            "value": "[parameters('tenantId')]"
        },
        "AZURE_CLIENT_ID": {
            "type": "string",
            "value": "[parameters('testApplicationId')]"
        },
        "AZURE_CLIENT_SECRET": {
            "type": "string",
            "value": "[parameters('testApplicationSecret')]"
        },
...
```

Comment (danieljurek) [reply sdk/eventhubs/tests.yml]: Update the `EnvVars` section of the tests.yml file: 

```
EnvVars:
    AZURE_TEST_MODE: RECORD
    AZURE_LOG_LEVEL: 4
```

All of the rest of the mappings are redundant now.

File: sdk/eventhubs/test-resources.json [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+            "metadata": {
+                "description": "The application client ID used to run tests."
+            }
+        },
+        "testApplicationSecret": {
+            "type": "string",
+            "metadata": {
+                "description": "The application client secret used to run tests."
+            }
+        },
```
Comment (danieljurek) [inline sdk/eventhubs/test-resources.json]: This closes the `testApplicationSecret` object, but not the `parameters` object, so the variables section looks like another parameter to the script. 

This line should be `}},` to close the `testApplicationSecret` object and close the `parameters` section.

**Pull request merged**